### DISCUSSION TITLE
Remove/Filter out some columns possible?

### DISCUSSION THREAD

**Discussion by G0nz0uk** [grafana/grafana-csv-datasource#42] [Q&A] [answered]
Hello,

I have to import a huge CSV that a system dumps out that we can't change, but I only need maybe 10 columns from it.

Is there a way I can just show these and not the whole CSV?

Thanks

**Comment by marcusolsson** [comment]
Yes! The caveat is that you need to define a schema with the 10 columns you want to keep. Once you've done that, click **Ignore unknown**, and any fields that's not defined in the schema will be removed.

**Reply by G0nz0uk** [reply]
Hi Marcus, so that is done in the panel itself and not the data source?

**Comment by G0nz0uk** [comment, marked as answer]
Is this where I do it?

![image](https://user-images.githubusercontent.com/15994497/109396122-efdf3400-7927-11eb-9864-dde5bbcd4a81.png)

**Reply by marcusolsson** [reply]
1. In the query editor, add the fields that you want to include, and optionally their types
2. Enable **Ignore unknown** to hide any field you have included

![Screenshot 2021-02-28 at 07 38 55@2x](https://user-images.githubusercontent.com/8396880/109410102-1e94f300-7998-11eb-9a61-1accccd1798f.png)

**Reply by G0nz0uk** [reply]
Amazing, thanks works perfectly.

One last question, can I rename a column name, for example if it's pulled in from the CSV as 'Country' can this be renamed as something else or does the CSV need to be edited?  Problem we have a 3rd party system exports the csv so we can't easily change it on the source end.

**Reply by marcusolsson** [reply]
Definitely! You can rename (and reorder) fields from any data source using the **Organize fields** transform.

**Reply by G0nz0uk** [reply]
Amazing! Thanks.
